packages/p2p-v2/src/components/PaymentMethodForm/PaymentMethodFormInput/PaymentMethodFormInput.tsx

```diff
@@ -0,0 +1,101 @@
+import React from 'react';
+import { THooks, TSelectedPaymentMethod } from 'types';
+import { Dropdown } from '@/components';
+import { TFormState } from '@/reducers/types';
+import { Button, Input, Text } from '@deriv-com/ui';
+import CloseCircle from '../../../public/ic-close-circle.svg';
+
+type TPaymentMethodFormInputProps = {
+    actionType: TFormState['actionType'];
+    availablePaymentMethods?: THooks.PaymentMethods.Get;
+    availablePaymentMethodsList: { text: string; value: string }[];
+    onAdd: (selectedPaymentMethod?: TSelectedPaymentMethod) => void;
+    reset: () => void;
+    selectedPaymentMethod: TFormState['selectedPaymentMethod'];
+};
+
+const PaymentMethodFormInput = ({
+    actionType,
+    availablePaymentMethods,
+    availablePaymentMethodsList,
+    onAdd,
+    reset,
+    selectedPaymentMethod,
+}: TPaymentMethodFormInputProps) => {
+    if (selectedPaymentMethod) {
+        // TODO: Remember to translate this
+        return (
+            <Input
+                defaultValue={selectedPaymentMethod?.display_name}
+                disabled
+                label='Choose your payment method'
+                rightPlaceholder={
+                    actionType === 'EDIT' ? null : (
+                        <CloseCircle
+                            className='p2p-v2-payment-method-form__icon--close'
+                            data-testid='dt_p2p_v2_payment_methods_form_close_icon'
+                            fill='#999999'
+                            height={15.7}
+                            onClick={() => {
+                                onAdd();
+                                reset();
+                            }}
+                            width={15.7}
+                        />
+                    )
+                }
+            />
+        );
+    }
+    return (
+        <>
+            <Dropdown
+                label='Payment method'
+                list={availablePaymentMethodsList}
+                name='Payment method'
+                onSelect={(value: string) => {
```

Comment:
We could separate this into a function for more readability, feels like its getting bigger here
```suggestion
                onSelect={onSelectPaymentMethod}
```